### PR TITLE
Move Http Impl

### PR DESCRIPTION
**Pull request by thurstonsand**
Moves the HTTP impl of `Transport` to its own top level module. Also, moves `ByteArrStream` out of `net` and into `helpers`

### CODE REVIEW

File: src/helpers/mod.rs [hunk 1]
```diff
@@ -1,9 +1,8 @@
 pub mod messaging;
 pub mod network;
-pub mod old_http;
-#[deprecated(note = "Use `Transport` instead")]
-pub mod old_network;
-/// TODO: the whole http module need to be moved to `net`
+// pub mod old_http;
+// #[deprecated(note = "Use `Transport` instead")]
+// pub mod old_network;
```
Comment (akoshelev) [inline src/helpers/mod.rs:5]:
```suggestion
```

File: src/http/server/handlers/create_query.rs [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 };
 use axum::{routing::post, Extension, Json, Router};
 use hyper::{Body, Request};
 use tokio::sync::{mpsc, oneshot};
 
-#[cfg_attr(feature = "enable-serde", derive(serde::Deserialize))]
-struct CreateQueryBody {
-    helper_positions: [HelperIdentity; 3],
-}
+// TODO: do we need this?
```
Comment (akoshelev) [inline src/http/server/handlers/create_query.rs:12]: if it is not used, probably not?

**Comment by akoshelev** [comment]
can you move `http` module to `net`?

File: src/http/server/handlers/query_input.rs [hunk 3]
```diff
@@ -34,9 +33,9 @@ async fn handler(
...
     let input_stream = RequestParts::new(req)
-        .extract::<ByteArrStream>()
-        .await
-        .map_err(|_| Error::BadQueryString("TODO: move ByteArrStream to the right package".into()))?
+        .extract::<ByteArrStreamFromReq>()
+        .await?
+        .0
         .and_then(|bytes| futures::future::ok(bytes.to_vec()))
         .map_err(TransportError::from);
     let query_input = QueryInput {
```
Comment (akoshelev) [inline src/http/server/handlers/query_input.rs:41]: I think we need another todo here. `FieldType` should not be sent twice per query. RC initiates the query telling us what field type to use and then submits the input assuming we know the field type already

**Review by akoshelev — CHANGES_REQUESTED** [review]

File: src/http/server/handlers/query_input.rs (hunk 3 above)
Comment (thurstonsand) [reply src/http/server/handlers/query_input.rs:41]: right. this is one of the points i wanted to discuss over VC. i have a proposed solution and wanted to pass it by you

File: src/http/server/handlers/create_query.rs (hunk 2 above)
Comment (thurstonsand) [reply src/http/server/handlers/create_query.rs:12]: ok. it will be easy enough to add back if we ever do need it

**Comment by thurstonsand** [comment]
i can move `http` to `net`, but let me do that in the last PR in the stack: https://github.com/thurstonsand/raw-ipa/pull/12 since I have made changes to the `http` folder along the way. easier to resolve diff

File: src/http/server/handlers/query_input.rs (hunk 3 above)
Comment (thurstonsand) [reply src/http/server/handlers/query_input.rs:41]: this field has been removed in the last PR: https://github.com/thurstonsand/raw-ipa/pull/12 (see commit: https://github.com/thurstonsand/raw-ipa/pull/12/commits/dd1fac7d5fe15ca8d00373b45193fb656226c779)

**Pull request merged**